lib/app.js

```diff
@@ -160,29 +273,142 @@ class AppView extends React.Component {
     );
   }
 
-  platform(value) {
-    return Platform.OS === value
-  }
-
   onNetworkAlertClosed() {
     setTimeout(() => this.networkAlertOpen = false, 500)
   }
 
+  /**
+   * Called when the user refreshes the list
+   * via a pull-to-refresh gesture.
+   */
   onRefresh() {
     debug('on refresh');
     track.pullRefresh();
     this.stopScanning();
-    actions.clearItems();
+    this.props.clearItems();
     this.startScanning();
   }
+
+  /**
+   * Called when the navigator finishes
+   * a transition to a new 'scene'.
+   */
+  onNavigated({ type }) {
+    debug('navigated', type);
+    track.view(type);
+  }
+
+  /**
+   * Called when a list-item is pressed.
+   *
+   * Depending on what prefs are set, this
+   * could open the expanded view, or simply
+   * navigate to the item's url.
+   */
+  onItemPress(item) {
```

Comment:
Have the general feeling that now app.js is a huge controller for all the views, a bit mixed for me, but won't block on this. We can revisit if we find a better way.